Source/Core/Core/PowerPC/SignatureDB/MEGASignatureDB.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Core/PowerPC/PPCSymbolDB.h"
+
+struct MEGASignatureReference
+{
+  u32 offset;
+  std::string name;
+
+  MEGASignatureReference(u32 _offset, std::string _name) : offset(_offset), name(_name) {}
+};
+
+struct MEGASignature
+{
+  std::vector<u32> code;
+  std::string name;
+  std::vector<MEGASignatureReference> refs;
+};
+
+class MEGASignatureDB
+{
+public:
+  MEGASignatureDB() = default;
+  ~MEGASignatureDB() = default;
+
+  bool Load(const std::string& file_path);
+  void Apply(PPCSymbolDB* symbol_db) const;
```

Comment:
PPCSymbolDB can be forward declared, eliminating the need for the PPCSymbolDB include in the header.